driver-sync/src/main/com/mongodb/client/internal/ClientSessionImpl.java

```diff
@@ -81,6 +87,8 @@ public void startTransaction() {
 
     @Override
     public void startTransaction(final TransactionOptions transactionOptions) {
+        checkServerVersionForTransactionSupport();
```

Comment:
Is there any reason to leave this here still?  Does the CommandMessage change cover us?  The behavior is a bit different since the exception will be thrown on first use, but I think that's ok.

Reply:
I reverted ClientSessionImpl#startTransaction and the tests passed. `./gradlew check` also passed locally.

Reply:
Starting an Evergreen patch: https://evergreen.mongodb.com/version/5c4911500ae6065de57e95d8